walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,105 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  /**
+   * This algorithm creates a List of all the times at least one attendee is busy, then uses that
+   * List to find all the times that everyone is available The worst-case runtime is O(n^2)
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendeesRequested = request.getAttendees();
+    int meetingDuration = (int) request.getDuration();
+
+    // check for no attendees
+    if (attendeesRequested.isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    // check duration of meeting request
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    List<TimeRange> badTimes = getBadTimes(events, attendeesRequested);
+
+    // list to hold all the available times (everything not in badTimes)
+    List<TimeRange> goodTimes = getGoodTimes(badTimes, meetingDuration);
+    return goodTimes;
+  }
+
+  /** Returns a List with all the times that requested attendees have conflicts. */
+  private List<TimeRange> getBadTimes(
```

Comment:
nit: This (and getGoodTimes as well) would be better as static (e.g.: static private List<TimeRange> getBadTimes (...) ).

In general, if a class method doesn't use any instance variables/methods (which is to say it's stateless), we usually declare them static, so 1) you don't need an instance to use the methods, and 2) the reader knows the function doesn't  depend on any values of the object.

Reply:
This makes sense,  I added that in